File: QuickSort/TypeScript/quicksort.ts
function quicksort(arr: number[]): number[] {
  if (arr.length < 2) {
    return arr;
  }
  let pivot=arr[0];
  let less=arr.filter((i)=>{return i<pivot})
  let greater=arr.filter((i)=>{return i>pivot})
  return quicksort(less).concat([pivot], quicksort(greater));
}
let l=[5,68,58,9,2,4]
console.log(quicksort(l))

File: QuickSort/Python/quicksort.py
"""
quicksort is divided the arr into two greater than and lesser than [pivot]
and then quicksort those 2 arrays again 

"""

def quicksort(arr):
   
    #base case
    if len(arr) < 2:
        return arr
    pivot =arr[0]
    less=[i for i in arr[1:] if i < pivot]
    greater=[i for i in arr[1:] if i > pivot]

    return quicksort(less)+ [pivot] + quicksort(greater)
l=[1,5,6,9,8,4]
print(quicksort(l))